content/en/blog/_posts/2021-12-15-pod-security-admission-beta.md

```diff
@@ -0,0 +1,774 @@
+```
+
+### Configure Pod Security
+
+Policies are applied to a namespace via labels. These labels are as follows:
+   * `pod-security.kubernetes.io/<MODE>: <LEVEL>` (required to enable pod security)
+   * `pod-security.kubernetes.io/<MODE>-version: <VERSION>` (*optional*, defaults to latest)
+
+A specific version can be supplied for each enforcement mode. The version pins the policy to the version that was shipped as part of the Kubernetes release. Pinning to a specific Kubernetes version allows for deterministic policy behavior while allowing flexibility for future updates to Pod Security Standards. The possible <MODE(S)> are `enforce`, `audit` and `warn`.
+
+### When to use `warn`?
+
+The typical uses for `warn` are:
+
+* `warn` at the same level but a different version (e.g. pin `enforce` to *restricted+v1.23* and `warn` at *restricted+latest*)
+* `warn` at a stricter level (e.g. `enforce` baseline, `warn` restricted)
+
+It's not recommended to use `warn` for the exact same level+version of the policy as `enforce`. In the admission sequence, if `enforce` fails, the entire sequence fails before evaluating the `warn`.
+
+
+First, create a namespace called `verify-pod-security` if not created earlier. For the demo, `--overwrite` is used when labeling to allow repurposing a single namespace for multiple examples.
+
+```shell
+kubectl create namespace verify-pod-security
+```
+
+### Deploy demo workloads
+
+Each workload represents a higher level of security that would not pass the profile that comes after it.
+
+For the following examples, use the `busybox` container runs a `sleep` command for 1 million seconds (~11 days) or until deleted. Pod Security is not interested in the container, but rather the Kubernetes manifests and corresponding security values that are being admitted.
+
+### Privileged level and workload
+
+For the privileged pod, use the [privileged policy](/docs/concepts/security/pod-security-standards/#privileged). This allows the process inside a container to gain new processes (also known as "privilege escalation") and can be dangerous if untrusted.
+
+First, let's apply a restricted Pod Security level for a test.
+
+```shell
+# enforces a "restricted" security policy and audits on restricted
+kubectl label --overwrite ns verify-pod-security \
+  pod-security.kubernetes.io/enforce=restricted \
+  pod-security.kubernetes.io/audit=restricted
+```
+
+Next, try to deploy a privileged workload in the namespace.
+
+```shell
+cat <<EOF | kubectl -n verify-pod-security apply -f -
+apiVersion: v1
+kind: Pod
+metadata:
+  name: busybox-privileged
+spec:
+  containers:
+  - name: busybox
```

Comment:
Note to myself: Tutorial is using `nginx` whereas blog is using `busybox`. Switch over the tutorial to use `busybox`

Reply:
I started using `nginx` and the comment came up that we should use something community owned, like `k8s.gcr.io/pause`, but I ran into a couple snags getting the demo to work smoothly.

I switched back to `nginx`, then later ran into issues when I was applying to the restricted profile (e.g: nginx is built / runs as root, so in my demo, I could either use `nginxinc/nginx-unprivileged` (not preferred) or something less opinionated about the runAsUser so I went with `busybox` + `sleep`. I'm not committed to using `busybox` if there's a compelling reason to use something else!

Reply:
Oh thanks for the history on why you switched to `busybox` from `nginx`. After reading that, makes more sense to continue to use `busybox`!!